streams/runners/storm/runtime/src/main/java/com/hortonworks/streamline/streams/runtime/storm/bolt/StreamlineTimestampExtractor.java

```diff
@@ -18,9 +18,14 @@ public StreamlineTimestampExtractor(String fieldName) {
     public long extractTimestamp(Tuple tuple) {
         StreamlineEvent event = (StreamlineEvent) tuple.getValueByField(StreamlineEvent.STREAMLINE_EVENT);
         Object ts = event.get(fieldName);
-        if (ts == null || !(ts instanceof Long)) {
+        if (ts == null) {
+            throw new IllegalArgumentException("Streamline event does not contain field: " + fieldName);
+        }
+
+        if (!(ts instanceof Number)) {
```

Comment:
The timestamp has to be "milliseconds since January 1, 1970 UTC", so by putting values other than long (byte, short, int) we cannot represent dates beyond Jan 1970. So IMO, better to retain the `instanceof Long` and may be we can explicitly state something like `Streamline event does not contain a long value in the field` in the exception. 
Recently there was a case where the customer was passing seconds (instead of millis) in a long field which was causing issues. Will be great if we can catch that also. E.g. If value is less than 2 Billion log a WARNING.

Reply:
Ok missed your comment
> it didn't work with test topology with short value of field (not strictly long)

Is it due to the way the values are parsed out of the test input JSON? Still think we should restrict to "long" type for production use cases.

Reply:
Yes we just read the json via JSON parser without validating/converting to conform the schema. If we want to maintain this to only allow long (type), schema validator/converter maybe needed as well.

Reply:
Ok in the real mode we allow only long types (based on schema) to be selected as timestamp fields. I am not sure if we should relax the real mode types to allow any Number types as a workaround for test mode.  I think this may be an issue in general for other processors that receives number types as well.

In the test spout is it possible to do some fix to promote/cast the byte/short/int to long (or the actual schema type) based on the output schema of the source ?

Reply:
@arunmahadevan Yes I just reverted the change from StreamlineTimestampExtractor and implemented converter which converts the value being conformed to the type of field.